cmd/cstor-pool-mgmt/pool/pool.go

```diff
@@ -120,16 +158,21 @@ func createPoolBuilder(cStorPool *apis.CStorPool) []string {
 
 // CheckValidPool checks for validity of CStorPool resource.
 func CheckValidPool(cStorPool *apis.CStorPool) error {
+	diskList := cStorPool.Spec.Disks.DiskList
+	poolType := cStorPool.Spec.PoolSpec.PoolType
 	if len(string(cStorPool.ObjectMeta.UID)) == 0 {
 		return fmt.Errorf("Poolname/UID cannot be empty")
 	}
-	if len(cStorPool.Spec.Disks.DiskList) < 1 {
+	if len(diskList) < 1 {
 		return fmt.Errorf("Disk name(s) cannot be empty")
 	}
-	if cStorPool.Spec.PoolSpec.PoolType == "mirrored" &&
-		len(cStorPool.Spec.Disks.DiskList)%2 != 0 {
+	if poolType == mirrored && len(diskList)%2 != 0 {
 		return fmt.Errorf("Mirror poolType needs even number of disks")
 	}
+	if poolType == raidz || poolType == raidz2 {
```

Comment:
As per Idiomatic Maya principles these can be designed as Predicates inside `pkg/cstorpool/v1alpha1/cstorpool.go`
e.g.
```go
func Predicate(csp *apis.CstorPool) bool

func IsRaid() Predicate {
  return func(csp *apis.CstorPool) bool {
    return csp.poolType == raidz || csp.poolType = =raidz2
  }
}
```

cc @sonasingh46 @utkarshmani1997